common/rpc/interceptor/slow_request_logger.go

```diff
@@ -0,0 +1,74 @@
+package interceptor
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/rpc/interceptor/logtags"
+	"go.temporal.io/server/common/tasktoken"
+	"google.golang.org/grpc"
+)
+
+// Certain types of methods are ignored as a rule.
+var ignoredMethodSubstrings = []string{"Poll", "GetWorkflowExecutionHistory"}
+
+type SlowRequestLoggerInterceptor struct {
+	logger       log.Logger
+	workflowTags *logtags.WorkflowTags
+	dc           *dynamicconfig.Collection
+}
+
+func NewSlowRequestLoggerInterceptor(
+	logger log.Logger,
+	dc *dynamicconfig.Collection,
```

Comment:
don't take the whole Collection here, take a dynamicconfig.DurationPropertyFn (just `func () time.Duration`). that makes it easier to test and also more efficient (by a tiny bit).

you can use dynamicconfig.GetDurationPropertyFn to create a static one for tests